helpers/utils.js

```diff
@@ -1,19 +1,238 @@
+const requestModule = require('request-promise-native');
+const request = requestModule.defaults({ 'proxy': 'http://proxyout.reform.hmcts.net:8080' });
+const fs = require('fs');
+
+const logger = Logger.getLogger('helpers/utils.js');
+
 const getSolicitorLoginDetails = () => {
     if (!process.env.CCD_E2E_PASSWORD) {
         throw new Error('You need to set CCD_E2E_EMAIL and CCD_E2E_PASSWORD env variables');
     }
     return {
-        username: 'michael.osinloye@hmcts.net' ,
+        username: process.env.CCD_E2E_EMAIL,
         password: process.env.CCD_E2E_PASSWORD
     };
 }
 
+const getCaseWorkerLoginDetails = () => {
+    if (!process.env.CCD_E2E_PASSWORD) {
+        throw new Error('You need to set CCD_E2E_EMAIL and CCD_E2E_PASSWORD env variables');
+    }
+    return {
+        username: process.env.CCD_CASEWORKER_E2E_EMAIL,
+        password: process.env.CCD_CASEWORKER_E2E_PASSWORD
+    };
+}
+
+async function getUserToken() {
+  logger.info('Getting User Token');
+
+  // Setup Details
+  const username = process.env.CCD_CASEWORKER_E2E_EMAIL;
+  const password = process.env.CCD_CASEWORKER_E2E_PASSWORD;
+  const redirectUri = 'https://div-pfe-aat.service.core-compute-aat.internal/authenticated';
+  const idamClientSecret = process.env.IDAM_CLIENT_SECRET;
+
+  const idamBaseUrl = 'https://preprod-idamapi.reform.hmcts.net:3511';
+
+  const idamCodePath = `/oauth2/authorize?response_type=code&client_id=divorce&redirect_uri=${redirectUri}`;
+
+  const codeResponse = await request.post({
+    uri: idamBaseUrl + idamCodePath,
+    headers: {
+      Authorization: 'Basic ' + Buffer.from(`${username}:${password}`).toString('base64'),
+      'Content-Type': 'application/x-www-form-urlencoded'
+    }
+  }).catch(error => {
+      console.log(error);
+  });
+
+  const code = JSON.parse(codeResponse).code;
+
+  const idamAuthPath = `/oauth2/token?grant_type=authorization_code&client_id=divorce&client_secret=${idamClientSecret}&redirect_uri=${redirectUri}&code=${code}`;
+  const authTokenResponse = await request.post({
+    uri: idamBaseUrl + idamAuthPath,
+    headers: {
+      'Content-Type': 'application/x-www-form-urlencoded'
+    }
+  });
+
+  logger.debug(JSON.parse(authTokenResponse)['access_token']);
+
+  return JSON.parse(authTokenResponse)['access_token'];
+}
+
+async function getUserId(authToken) {
+  logger.info('Getting User Id');
+
+  const idamBaseUrl = 'https://preprod-idamapi.reform.hmcts.net:3511';
+
+  const idamDetailsPath = '/details';
+  const userDetails = await request.get({
+    uri: idamBaseUrl + idamDetailsPath,
+    headers: {
+      Authorization: `Bearer ${authToken}`
+    }
+  });
+
+  logger.debug(JSON.parse(userDetails).id);
+
+  return JSON.parse(userDetails).id;
+}
+
+async function getServiceToken() {
+  logger.info('Getting Service Token');
+
+  const serviceSecret = process.env.SERVICE_SECRET;
+
+  const s2sBaseUrl = 'http://rpe-service-auth-provider-aat.service.core-compute-aat.internal';
```

Comment:
Should we externalise these URLs to config?

Reply:
Can certainly, but right now this is a nightly e2e test purely for AAT. There is currently no intention to run these on any other envs right now.

We can definitely make these better though, but its priority driven I think.